core/node/crypto/chain_monitor.go

```diff
@@ -255,19 +278,22 @@ func (ecm *chainMonitor) RunWithBlockPeriod(
 				callbacksExecuted   sync.WaitGroup
 			)
 
+			ecm.mu.Lock()
+			fromBlock := ecm.fromBlock
+			fromBlockVersion := ecm.fromBlockVersion
+			ecm.mu.Unlock()
+
 			// ensure that the search range isn't too big because RPC providers
 			// often have limitations on the block range and/or response size.
-			if head.Number.Uint64()-fromBlock.Uint64() > 25 {
+			if head.Number.Cmp(fromBlock) > 0 && head.Number.Uint64()-fromBlock.Uint64() > 25 {
```

Comment:
nit: is this condition redundant? header block # is already > fromBlock if it's at least fromBlock + 25, yes?

Reply:
2 reasons:
- in theory it is possible to subscribe on events from future blocks. In that case fromBlock > head and therefore the subtraction underflows (go doesn't panic on int over/underflows)
- head is retrieved externally from an rpc provider. If for whatever reason the rpc node reverted a chain segment head can become < fromBlock -> same problem with the subtraction underflow.